include/envoy/http/filter.h

```diff
@@ -193,6 +193,9 @@ class StreamFilterCallbacks {
   virtual const ScopeTrackedObject& scope() PURE;
 };
 
+using RouteConfigUpdatedCallback = std::function<void(bool)>;
+using RouteConfigUpdatedCallbackSharedPtr = std::shared_ptr<RouteConfigUpdatedCallback>;
```

Comment:
Can you add a comment explaining the lifetime issues here? I.e. who has to hold the strong pointer, how weak pointer is used internally.

Reply:
done